src/controls/TextControl.tsx

```diff
@@ -95,26 +96,69 @@ function TextControlInput({
   textControlOptions,
   ...rest
 }: TextControlInputProps) {
+  const { inputProps, ...restTextControlOptions } = textControlOptions
+
   if (
-    !(`type` in textControlOptions) ||
-    textControlOptions.type === undefined
+    !(`type` in restTextControlOptions) ||
+    restTextControlOptions.type === undefined
   ) {
-    return <Input {...{ ...rest, ...textControlOptions }} />
+    return (
+      <Input
+        {...{
+          ...rest,
+          ...restTextControlOptions,
```

Comment:
TextControlOptions isn't intended to overlap with the props on an input, so it shouldn't be spread into the Input here

Reply:
oh makes total sense!